### PR TITLE
CTCTRADERS-1337R1: Redirect to task list page if dependent section is incomplete

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: app/controllers/actions/CheckDependentSectionAction.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+class CheckDependentSectionCompletionAction(val dependentSection: DependentSection, implicit val executionContext: ExecutionContext)
+    extends ActionFilter[DataRequest] {
+  override protected def filter[A](request: DataRequest[A]): Future[Option[Result]] = {
+    val reader: UserAnswersReader[_] = DependentSection.dependentSectionReader(dependentSection, request.userAnswers)
+    if (reader.run(request.userAnswers).isDefined) {
+      Future.successful(None)
+    } else {
+      Future.successful(Some(Redirect(routes.DeclarationSummaryController.onPageLoad(request.userAnswers.id))))
```
Comment (amaalali) [inline app/controllers/actions/CheckDependentSectionAction.scala:36]: Should we add an info log here and we can info to see where they were trying to go (I think we can get this off the request)?

**Review by amaalali — COMMENTED** [review]
The implementation looks good but I think that we need a test for `CheckDependentSectionCompletionAction`. Also it would be good to get some logging when the filter action does run

**Review by amaalali — APPROVED** [review]

**Comment by amaalali** [comment]
I totally missed the test you added for  `CheckDependentSectionCompletionAction`! With the logging you just added it all looks good

**Comment by ashwinirudrappa** [comment]
test this please

**Pull request merged**